FridgeExercise/Refrigerator.cs

```diff
@@ -0,0 +1,249 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace FridgeExercise
+{
+	internal class Refrigerator
+	{
+		public int RefrigeratorID { get; set; }
+		private int CounterID = 1;
+		public string Model { get; set; }
+		public string Color { get; set; }
+		public int NumberOfShelves { get; set; }
+		public List<Shelf> Shelves { get; set; }
+
+
+		public Refrigerator(string model, string color, int numberOfShelves) {
+			RefrigeratorID = CounterID;
+			CounterID++;
+			Model = model;
+			Color = color;
+			NumberOfShelves = numberOfShelves;
+			Shelves = new List<Shelf>();
+
+		}
+
+		public string ToString(Refrigerator refrigerator)
+		{
+			return "Refrigerator: Model:  " + refrigerator.Model + "  Color: " + refrigerator.Color + "  NumberOfShelves:  "+ refrigerator.NumberOfShelves + "";
+		}
+
+		public double SpaceIsLeftInTheFridge(List<Shelf> shelves)
+		{
+			double currentSpace = 0;
+			foreach (Shelf shelf in shelves)
+			{
+				currentSpace += shelf.CurrentSpaceInShelf;
+			}
+			return currentSpace;
+		}
+		public void PutItemInTheFridge(Item item, List<Shelf> shelves)
+		{
+			foreach (Shelf shelf in shelves)
+			{
+				if (shelf.CurrentSpaceInShelf - item.SpaceItTakes >= 0)
+				{
+
+					shelf.CurrentSpaceInShelf -= item.SpaceItTakes;
+					shelf.Items.Add(item);
+					item.ItemBelongsToShelf = shelf;
+					return;
+				}
+			}
+			Console.WriteLine("There is no  space for this item");
+		}
+		public string TakeOutItem(int ItemId, List<Shelf> shelves)
+		{
+			Item tempItem=null;
+			foreach(Shelf shelf in shelves)
+			{
+				foreach(Item item in shelf.Items.ToList())
```

Comment:
```suggestion
				foreach (Item item in shelf.Items))
```

Styling + no need for the ToList method, it's already a list